docs/WIRELESS.md

```diff
@@ -158,6 +158,258 @@ The fact that EVE itself stays out of Firmware upgrade business means that you m
+state, EVE will start periodically making a POST request to the [Radio endpoint](../api/PROFILE.md#Radio).
+If this (optional) endpoint is not implemented, the default policy for radio transmission
+will remain in effect. If the endpoint is available, EVE will provide an update of the
+current state of wireless devices in the POST request body, formatted and marshalled using
+the [RadioStatus](./proto/profile/local_profile.proto) proto message.
+
+If a response from the application contains no content (response code 204), EVE assumes
+that the intended radio silence state has not changed (initial intended state is a disabled
+radio silence).
+Application response with non-empty content (response code 200) is unmarshalled into
+[RadioConfig](../api/proto/profile/local_profile.proto) protobuf message. If the unmarshalling
+succeeded and the token matches the expected value configured through the controller,
+EVE will accept the new radio configuration. Currently, apart from the token, RadioConfig
+contains only a single boolean field which determines if the radio silence should be imposed
+or lifted.
+
+Whenever the newly received intended state of the radio silence mode differs from
+the currently applied state, EVE will trigger an operation of switching radio transmission
+ON/OFF on all (used) wireless devices. This operation takes some time, during which EVE
+stops publishing radio status updates and receiving new radio configs. This is purely
+to simplify the implementation and avoid any interleaving between config updates and state
+changes. Once the new radio config was applied (successfully or with an error),
+EVE will restart the periodic radio status updates and will POST the outcome of the operation
+back to the application via the radio endpoint.
+
+A formal definition for the syntax and the semantics of the radio endpoint can be found
+[here](../api/PROFILE.md#Radio)
+
+#### Persistence
+
+The intended radio configuration received from Local profile server is persisted
+by EVE (under `/persist` partition) and remains in effect even if the application
+restarts or the edge node reboots (but please see [risks and limitations](#risks-and-limitations)
+associated with the edge node reboot).
+Only once the Local profile server is un-configured or un-deployed will EVE discard
+the last radio config and restore the default behavior of disabled radio silence.
+
+#### Controller connectivity
+
+While the radio silence is imposed, it is perfectly normal for the edge node to lose access
+to the controller, especially if there are no wired connectivity alternatives.
+Normally, this could cause the network configuration to fallback to a previous state
+in an attempt to restore the connectivity. However, in this case this behavior is not desirable,
+therefore the network connectivity testing and the whole fallback mechanism is disabled
+during the radio silence.
```

Comment:
I must have missed the code for this. Does it only apply when radios is the only management port(s)?

Reply:
The check is here: https://github.com/lf-edge/eve/blob/master/pkg/pillar/devicenetwork/dnc.go#L129-L132
Note that the testing & fallback are disabled during temporary radio silence even if there are ethernet management ports. Given that radio silence is used for safety-critical applications, I believe it is wise to keep the device still and prevent it from changing the internal state. Even if there is a wired connectivity option, it might not be working (and only wireless does), which would trigger fallback during radio silence. Also, it is quite common that an ethernet port actually leads to a satellite modem, which will be also turned off during radio silence by an operator (outside of eve management).

Reply:
But the getconfig task in zedagent is the one which will give up (after 5 minutes if the update is in progress and after 7 days otherwise) causing a fallback in the former case. I think that code needs an added check to match what you documented.

Reply:
Correct.